packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -21,35 +22,43 @@ export const accordionHeaderShorthandProps: Array<keyof AccordionHeaderSlots> =
  * @param ref - reference to root HTMLElement of AccordionHeader
  */
 export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<HTMLElement>): AccordionHeaderState => {
+  const {
+    icon,
+    button,
+    children,
+    expandIcon,
+    inline = false,
+    size = 'medium',
+    expandIconPosition = 'start',
+    ...rest
+  } = props;
   const { onHeaderClick: onAccordionHeaderClick, disabled, open } = useAccordionItemContext();
-  const id = useId('accordion-header-', props.id);
-  const innerRef = React.useRef<HTMLElement>(null);
 
-  const buttonShorthand = useARIAButton(props.button, {
+  const buttonShorthand = useARIAButton(button, {
     required: true,
     defaultProps: {
-      id,
       disabled,
-      // 'aria-controls': panel?.id,
     },
   });
 
   return {
-    ...props,
     disabled,
     open,
-    ref: useMergedRefs(ref, innerRef),
-    size: props.size || 'medium',
-    inline: props.inline || false,
-    role: props.role || 'heading',
-    expandIconPosition: props.expandIconPosition || 'start',
+    size,
+    inline,
+    expandIconPosition,
     components: {
       root: 'div',
       button: 'button',
       expandIcon: AccordionHeaderExpandIcon,
     },
-    icon: resolveShorthand(props.icon),
-    expandIcon: resolveShorthand(props.expandIcon, {
+    root: getNativeElementProps('div', {
+      ref,
+      role: 'heading',
+      ...rest,
```

Comment:
```suggestion
      ...props,
```

Let's keep spreading `props`, an example is custom handling for `type` in `Button`.

```tsx
const { type, ...rest } = props
const root = getNativeElementProps('div', { ...rest, ref }) // Oops, "type" will not go anymore to DOM
```